### PR TITLE
Fix CVEs, use scratch based Image

### PR DESCRIPTION
**Pull request by krishgobinath**
## Description
Build scratch based Image, with only statically built applications[calicoctl & calicoctl-wait]
1) Built small application to wait for in an indefinite loop.

### CODE REVIEW

File: Dockerfile.amd64 [hunk 1]
```diff
@@ -24,9 +24,7 @@ LABEL name="Calico CLI tool" \
       maintainer="maintainers@projectcalico.org"
 
 ADD bin/calicoctl-linux-amd64 /calicoctl
-
-RUN mkdir /licenses
```
Comment (robbrockbank) [inline Dockerfile.amd64:28]: I'm not sure what this was originally for so I've no idea if removing it is sensible.  Do you have any more details on this?

File: calicoctl-wait/calicoctl-wait.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
+// Copyright (c) 2020 Tigera, Inc. All rights reserved.
```
Comment (robbrockbank) [inline calicoctl-wait/calicoctl-wait.go:1]: What is this for?

File: Dockerfile.amd64 (hunk 1 above)
Comment (krishgobinath) [reply Dockerfile.amd64:28]: I don't know why that file is required. I'll bring back the original file.

File: calicoctl-wait/calicoctl-wait.go (hunk 2 above)
Comment (krishgobinath) [reply calicoctl-wait/calicoctl-wait.go:1]: The calico-wait tool is used to replace Linux utils used in the container image and go for a pure scratch image.

File: calicoctl-wait/calicoctl-wait.go [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+package main
+
+import (
+	"time"
+)
+
+func main() {
+	// Spin continously in tight loop
+	for {
+		time.Sleep(500 * time.Millisecond)
```
Comment (doublek) [inline calicoctl-wait/calicoctl-wait.go:24]: This seems too low, should we just sleep longer?

Comment (bcreane) [reply calicoctl-wait/calicoctl-wait.go:24]: @krishgobinath - I prematurely merged - didn't notice that @doublek was pending. Can you open another PR to address this comment?

Comment (krishgobinath) [reply calicoctl-wait/calicoctl-wait.go:24]: I had unit tested with 500ms and 1 second, I didn't spot any difference between those 2 in terms of CPU utilization rate. The worst case is if we don't sleep then CPU utilization rate is 100%.

Comment (krishgobinath) [reply calicoctl-wait/calicoctl-wait.go:24]: @bcreane If required I'll do that.

Comment (bcreane) [reply calicoctl-wait/calicoctl-wait.go:24]: I chatted with @doublek - he's okay skipping this change. So ... never mind.

**Pull request merged**